nixos/modules/services/mail/mailman.nix

```diff
@@ -0,0 +1,120 @@
+{ config, pkgs, lib, ... }:          # mailman.nix
+
+with lib;
+
+let
+
+  cfg = config.services.mailman;
+
+  pythonEnv = pkgs.python3.withPackages (ps: [ps.mailman]);
+
+  mailmanExe = with pkgs; stdenv.mkDerivation {
+    name = "mailman-" + python3Packages.mailman.version;
+    unpackPhase = ":";
+    installPhase = ''
+      mkdir -p $out/bin
+      sed >"$out/bin/mailman" <"${pythonEnv}/bin/mailman" \
+        -e "2 iexport MAILMAN_CONFIG_FILE=/etc/mailman.cfg"
```

Comment:
If this needs to work for any user as you mentioned in https://github.com/NixOS/nixpkgs/pull/67461#discussion_r319168331, then I'd think `environment.sessionVariables` plus `systemd.services.mailman.environment` would be much cleaner

Reply:
I don't think that's cleaner at all. I don't want to pollute the global environment to set a variable that's relevant only for one, maybe two users on the system. This solution is much cleaner, IMHO.

Reply:
Alright yeah that sounds reasonable. There is however `makeWrapper ${pythonEnv}/bin/mailman $out/bin/mailman --set MAILMAN_CONFIG_FILE /etc/mailman.cfg` you can use to make such wrappers.

Reply:
I know. That file is a wrapper already, though, and I'd rather not wrap that wrapped script in another wrapper script.

Reply:
Because makeWrapper uses `exec` to start the binary, there's little runtime cost to doing this, and this way will continue to work in the future even if mailman stops being a wrapper, it also makes the code here clearer since one doesn't need to know sed and that mailman is a wrapper to see what this is doing.

Reply:
If you feel strongly about this issue, then please feel free to make a PR that changes the `mailman` script to the way you'd prefer. I won't object such a change, but I don't care enough to do it myself.

Reply:
I don't care that much about it too, but maybe something for the future :)